pkg/rpc/rpc_handler.go

```diff
@@ -98,7 +147,27 @@ func RunRPCHandler(policyReconciler *controllers.PolicyEndpointsReconciler) erro
 		return errors.Wrap(err, "network policy agent: failed to listen to gRPC port")
 	}
 	grpcServer := grpc.NewServer()
-	rpc.RegisterNPBackendServer(grpcServer, &server{policyReconciler: policyReconciler, log: rpcLog})
+
+	// Connect to metadata cache service
+	serviceIP, _ := utils.GetServiceIP(clientset, metadataServiceName, metadataServiceNamespace)
+	cacheClient, err := newCacheClient(serviceIP + ":50051")
+	if err != nil {
+		rpcLog.Error(err, "unable to connect to aws-k8s-metadata service, continuing without")
+		utils.CacheClientConnected = false
+	} else {
+		utils.CacheClientConnected = true
```

Comment:
If the CacheClientConnected is `false`, why do we need to still run `syncLocalCache` ?